packages/jbrowse-cli/src/commands/add-connection.ts

```diff
@@ -0,0 +1,317 @@
+            { exit: 110 },
+          )
+
+        break
+      }
+    }
+
+    if (!configContents.connections) {
+      configContents.connections = []
+    }
+    const idx = configContents.connections.findIndex(
+      configConnection => configConnection.connectionId === connectionId,
+    )
+
+    if (idx !== -1) {
+      this.debug(`Found existing connectionId ${connectionId} in configuration`)
+      if (runFlags.force || runFlags.overwrite) {
+        this.debug(`Overwriting connection ${connectionId} in configuration`)
+        configContents.connections[idx] = connectionConfig
+      } else
+        this.error(
+          `Cannot add connection with id ${connectionId}, a connection with that id already exists.\nUse --overwrite if you would like to replace the existing connection`,
+          { exit: 40 },
+        )
+    } else configContents.connections.push(connectionConfig)
+
+    this.debug(`Writing configuration to file ${configPath}`)
+    await fsPromises.writeFile(
+      configPath,
+      JSON.stringify(configContents, undefined, 2),
+    )
+
+    this.log(
+      `${idx !== -1 ? 'Overwrote' : 'Added'} connection "${name}" ${
+        idx !== -1 ? 'in' : 'to'
+      } ${configPath}`,
+    )
+  }
+
+  async checkLocation(location: string) {
+    let manifestJson: string
+    try {
+      manifestJson = await fsPromises.readFile(
+        path.join(location, 'manifest.json'),
+        {
+          encoding: 'utf8',
+        },
+      )
+    } catch (error) {
+      this.error(
+        'Could not find the file "manifest.json". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 50 },
+      )
+    }
+    let manifest: { name?: string } = {}
+    try {
+      manifest = JSON.parse(manifestJson)
+    } catch (error) {
+      this.error(
+        'Could not parse the file "manifest.json". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 60 },
+      )
+    }
+    if (manifest.name !== 'JBrowse') {
+      this.error(
+        '"name" in file "manifest.json" is not "JBrowse". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 70 },
+      )
+    }
+  }
+
+  async resolveURL(location: string, check = true) {
+    let locationUrl: URL | undefined
+    try {
+      locationUrl = new URL(location)
+    } catch (error) {
+      this.error('The location provided is not a valid URL', { exit: 80 })
+    }
+    if (locationUrl) {
+      let response
+      try {
+        if (check) {
+          response = await fetch(locationUrl, { method: 'HEAD' })
+        }
+        if (!response || response.ok) return locationUrl.href
+      } catch (error) {
+        // ignore
+        this.error('Unable to fetch from URL', { exit: 90 })
+      }
+    }
+    return this.error(`Could not resolve to a URL: "${location}"`, {
+      exit: 100,
+    })
```

Comment:
Could be good to add the response code if the response is not ok, and the error message if the fetch throws an error.